File: backend/components/metiers/administration_publique/utils/metrics/samples/sample_usage.test.js
// sample_usage.test.js
// Tests unitaires avancés pour les exemples metrics JS
const metrics = require('../core/metrics');
const data = require('./sample_metrics_data.json').metrics;

describe('Sample usage metrics JS', () => {
  it('calcule correctement la moyenne', () => {
    expect(metrics.calculerMoyenne(data)).toBeCloseTo(15.814, 2);
  });
  it('calcule correctement la médiane', () => {
    expect(metrics.calculerMediane(data)).toBe(3.5);
  });
  it('gère les tableaux vides', () => {
    expect(() => metrics.calculerMoyenne([])).toThrow();
  });
});

File: backend/components/metiers/administration_publique/utils/metrics/samples/sample_usage.test.py
# sample_usage.test.py
"""Tests unitaires avancés pour les exemples metrics Python"""
from ..core import metrics
import json
import pytest

def test_calculer_moyenne():
    with open('sample_metrics_data.json') as f:
        data = json.load(f)["metrics"]
    assert abs(metrics.calculer_moyenne(data) - 15.814) < 0.01

def test_calculer_mediane():
    with open('sample_metrics_data.json') as f:
        data = json.load(f)["metrics"]
    assert metrics.calculer_mediane(data) == 3.5

def test_moyenne_vide():
    with pytest.raises(Exception):
        metrics.calculer_moyenne([])
